ledger/participant-integration-api/src/main/scala/platform/store/dao/events/TransactionsReader.scala

```diff
@@ -439,14 +441,16 @@ private[dao] final class TransactionsReader(
           )
         }
       }
-      .mapConcat(identity)
       .async
       .mapAsync(eventProcessingParallelism) { raw =>
         Timed.future(
           metrics.daml.index.decodeStateEvent,
-          Future(ContractStateEventsReader.toContractStateEvent(raw, lfValueTranslation)),
+          Future.traverse(raw)(rawEvent =>
```

Comment:
The purpose here is to align the behavior to the other stream construction methods in this class. Indeed, this will be gone soon, but this change is small and an improvement over the current state so in the spirit of improving the current performance, I've applied this improvement here as well.